src/components/OperationTile/useGetOperationDestination.ts

```diff
@@ -0,0 +1,31 @@
+import { RawPkh } from "../../types/Address";
+import { useIsOwnedAddress } from "../../utils/hooks/getAccountDataHooks";
+
+export type OperationDestination = "incoming" | "outgoing" | "unrelated";
+
+/**
+ * This hook is used to determine if an operation is outgoing, incoming, or unrelated.
+ *
+ * It can be unrelated if a user initiated some operations through a contract call,
+ * but isn't a sender or a receiver of assets.
+ *
+ * @param sender - Who is marked as the operation's sender
+ * @returns OperationDestination
+ */
+export const useGetOperationDestination = (
+  sender: RawPkh | null | undefined,
+  receiver: RawPkh | null | undefined
+) => {
+  const isSenderOwned = useIsOwnedAddress(sender || "");
+  const isReceiverOwned = useIsOwnedAddress(receiver || "");
+
+  if (!isSenderOwned && !isReceiverOwned) {
+    return "unrelated";
+  }
+
```

Comment:
The logic seems to be simplified quite a lot?

- Maybe we should also handle separately the case when both sender and receiver are owned?

- For the account drawer the "current" account should take precedent for choosing the sign?

Reply:
when they both owned it's still an outgoing operation because you lose the fee
"current" account logic is too complicated and while it's nice in a way, it's also a bit confusing to see the same operation as incoming & outgoing in 2 different places